packages/sitecore-jss-nextjs/src/middleware/redirects-middleware.ts

```diff
@@ -86,6 +92,22 @@ export class RedirectsMiddleware extends MiddlewareBase {
         return res || NextResponse.next();
       }
 
+      // Find context site language and replace token
+      if (REGEXP_CONTEXT_SITE_LANG.test(existsRedirect.target)) {
+        const siteInfo = this.config.sites?.find(
+          (siteItem: SiteInfo) => siteItem.name === site?.name
+        );
+
+        if (!siteInfo) {
+          throw new Error(`Could not resolve context site language for site ${site.name}`);
+        }
+
+        existsRedirect.target = existsRedirect.target.replace(
+          REGEXP_CONTEXT_SITE_LANG,
+          siteInfo?.language
+        );
```

Comment:
I don't follow why we need this extra `siteInfo` variable and site lookup. Don't we already have this with the `site` variable (above)?
```suggestion
        existsRedirect.target = existsRedirect.target.replace(
          REGEXP_CONTEXT_SITE_LANG,
          site.language
        );
```

Reply:
I wanted to use this `siteInfo` but I need get language from config file because in `siteInfo` is property `language` but it's from route or default language(EN). SXA has context site language where you can set in Settings of site and this parameter is not related to routing
![image](https://user-images.githubusercontent.com/100142572/234816771-25138762-9a57-4642-9b24-2ee1b7c1fc2f.png)

Reply:
Ah. This is more a local development configuration problem. The resolved `site` is probably the default so has the default language (en). If you set the "emptysite" host name field in Sitecore to have "localhost", it'll then resolve to that one and have the da language you're expecting in the middleware. 

Check out the [docs SiteResolver section](https://doc.sitecore.com/xmc/en/developers/xm-cloud/the-next-js-multisite-add-on.html#:~:text=the%20sites%20property.-,SiteResolver,-The%20SiteResolver%20class) for a better explanation.

The middleware (and anywhere else that uses site resolver) needs to be able to trust / use the resolved site.